frontend/docsettings.lua

```diff
@@ -130,54 +134,51 @@ function DocSettings:getLastSaveTime(doc_path)
     end
 end
 
-function DocSettings:ensureSidecar(sidecar)
-    if lfs.attributes(sidecar, "mode") ~= "directory" then
-        lfs.mkdir(sidecar)
-    end
-end
-
 --- Opens a document's individual settings (font, margin, dictionary, etc.)
--- @string docfile path to the document (e.g., `/foo/bar.pdf`)
+-- @string doc_path path to the document (e.g., `/foo/bar.pdf`)
 -- @treturn DocSettings object
-function DocSettings:open(docfile)
-    --- @todo (zijiehe): Remove history_path, use only sidecar.
-
+function DocSettings:open(doc_path)
     -- NOTE: Beware, our new instance is new, but self is still DocSettings!
     local new = DocSettings:extend{}
-    new.history_file = new:getHistoryPath(docfile)
 
-    local sidecar = new:getSidecarDir(docfile)
-    new.sidecar = sidecar
-    DocSettings:ensureSidecar(sidecar)
-    -- If there is a file which has a same name as the sidecar directory,
-    -- or the file system is read-only, we should not waste time to read it.
-    if lfs.attributes(sidecar, "mode") == "directory" then
-        -- New sidecar file name is metadata.{file last suffix}.lua.
-        -- So we can handle two files with only different suffixes.
-        new.sidecar_file = new:getSidecarFile(docfile)
-        new.legacy_sidecar_file = sidecar.."/"..
-                                  ffiutil.basename(docfile)..".lua"
+    new.sidecar_dir_doc = new:getSidecarDir(doc_path, "doc")
+    new.sidecar_file_doc = new:getSidecarFile(doc_path, "doc")
+    local sidecar_file_doc, legacy_sidecar_file
+    if lfs.attributes(new.sidecar_dir_doc, "mode") == "directory" then
+        sidecar_file_doc = new.sidecar_file_doc
+        legacy_sidecar_file = new.sidecar_dir_doc.."/"..ffiutil.basename(doc_path)..".lua"
+    end
+    new.sidecar_dir_dir = new:getSidecarDir(doc_path, "dir")
+    new.sidecar_file_dir = new:getSidecarFile(doc_path, "dir")
+    local sidecar_file_dir
+    if lfs.attributes(new.sidecar_dir_dir, "mode") == "directory" then
+        sidecar_file_dir = new.sidecar_file_dir
```

Comment:
The current variable/key names, a bit inconsistent, makes this section a bit uneasy to read (as natural English):
```
sidecar_dir_doc
sidecar_file_doc
legacy_sidecar_file 
sidecar_dir_dir 
sidecar_file_dir 
history_file
```
May be this kind of naming would sound/read better?:
```
doc_sidecar_dir
doc_sidecar_file
legacy_sidecar_file 
dir_sidecar_dir
dir_sidecar_file
history_file
```